Occasional operator test failures

### Description

Recent prs seem to be related to some failed startup / migration https://github.com/keycloak/keycloak/actions/runs/5726457904/job/15517348959?pr=22152#step:8:2561 and https://github.com/keycloak/keycloak/actions/runs/5726457904/job/15517348959?pr=22152#step:8:2699 - this does not happen consistently, but once it does it seems to inhibit that keycloak instance from coming up.  Presumably this has to do with sharing the database instance for all tests in a test class.

The root exception message is 
```
Caused by: org.postgresql.util.PSQLException: ERROR: column "user_setup_allowed" of relation "authentication_execution" does not exist
```

This may have become more prevelent by changes to speed up the tests - the immediate termination of pods and not waiting for instances to become ready could cause a prior test to corrupt some expection of the database state.  However I'm not familar enough to know why it should even be entering migration logic.

cc @vmuzikar

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
@@ -265,7 +265,7 @@ public class BaseOperatorTest implements QuarkusTestAfterEachCallback {
           return;
       }
       try {
-          if (!context.getTestStatus().isTestFailed() && !(context.getTestStatus().getTestErrorCause() instanceof TestAbortedException)) {
+          if (!context.getTestStatus().isTestFailed() || context.getTestStatus().getTestErrorCause() instanceof TestAbortedException) {
               return;
           }
           Log.warnf("Test failed with %s: %s", context.getTestStatus().getTestErrorCause().getMessage(), context.getTestStatus().getTestErrorCause().getClass().getName());

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/WatchedSecretsTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/WatchedSecretsTest.java
@@ -22,6 +22,7 @@ import io.fabric8.kubernetes.api.model.SecretBuilder;
 import io.fabric8.kubernetes.api.model.apps.StatefulSet;
 import io.quarkus.logging.Log;
 import io.quarkus.test.junit.QuarkusTest;
+
 import org.awaitility.Awaitility;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
@@ -105,7 +106,7 @@ public class WatchedSecretsTest extends BaseOperatorTest {
 
     @Test
     public void testSecretsCanBeUnWatched() {
-        var kc = getTestKeycloakDeployment(true);
+        var kc = getTestKeycloakDeployment(false);
         deployKeycloak(k8sclient, kc, true);
 
         Log.info("Updating KC to not to rely on DB Secret");
@@ -230,7 +231,7 @@ public class WatchedSecretsTest extends BaseOperatorTest {
         kc.getSpec().getDatabaseSpec().setUsernameSecret(null);
         kc.getSpec().getDatabaseSpec().setPasswordSecret(null);
 
-        var username = new ValueOrSecret("db-username", "postgres");
+        var username = new ValueOrSecret("db-username", "kc-user");
         var password = new ValueOrSecret("db-password", "testpassword");
 
         kc.getSpec().getAdditionalOptions().remove(username);

--- a/operator/src/test/java/org/keycloak/operator/testsuite/utils/K8sUtils.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/utils/K8sUtils.java
@@ -30,7 +30,6 @@ import io.quarkus.logging.Log;
 
 import org.awaitility.Awaitility;
 import org.keycloak.operator.crds.v2alpha1.deployment.Keycloak;
-import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakSpecBuilder;
 import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakStatusCondition;
 
 import java.io.ByteArrayOutputStream;
@@ -83,10 +82,6 @@ public final class K8sUtils {
         if (deployTlsSecret) {
             set(client, getDefaultTlsSecret());
         }
-
-        // speed the cleanup of pods
-        kc.setSpec(new KeycloakSpecBuilder(kc.getSpec()).editOrNewUnsupported().editOrNewPodTemplate().editOrNewSpec()
-                .withTerminationGracePeriodSeconds(0L).endSpec().endPodTemplate().endUnsupported().build());
 
         set(client, kc);
 

